planning/behavior_path_planner/autoware_behavior_path_planner_common/include/autoware/behavior_path_planner_common/utils/utils.hpp

```diff
@@ -236,15 +248,38 @@ bool set_goal(
  */
 const Pose refineGoal(const Pose & goal, const lanelet::ConstLanelet & goal_lanelet);
 
+/**
+ * @brief Recreate the path with a given goal pose.
```

Comment:
The name of the function and its description seem a bit confusing. It recreates a path with a refined goal, but the function is named refine_path_with_goal.

I don’t have a better suggestion at the moment, but it’s fine to leave it as is unless someone comments on it later. Alternatively, should we consider moving it to the goal planner, since it seems to be used only by the goal planner?

@soblin @kosuke55, may I know what are your thoughts about this?

Reply:
I agree moving the function to goal planner. The `refineGoal` function seems to be used only by the goal planner too.